### ISSUE TITLE
[MA0042] Should not be raised, if synchronous method is called from within suggested async version

### ISSUE THREAD

**Issue by Pretasoc** [meziantou/Meziantou.Analyzer#788]
### Version of the Meziantou.Analyzer NuGet package

2.0.188

### Rule Identifier

MA0042

### Target Framework

net9.0

### C# Language version

preview

### Description

`MA0042` should exclude the method, that contains the call to the synchronous method from the list of candiates of asynchronous methods.

In the following example, the asynchronous method is required by an interface (and i don'T want to use an explicit implementation), but the actual implementation can run synchronously with just a simple try wrapper.

### Reproduction Steps

Minimal code:

```c#
#r "nuget: Meziantou.Analyzer, 2.0.188"

interface IAsyncFoo
{
    ValueTask BarAsync();
}

public class Foo : IAsyncFoo
{
    public ValueTask BarAsync()
    {
        try
        {
            Bar();
            return default;
        }
        catch(Exception e)
        {
            Task failure = Task.FromException(e);
            return new ValueTask(failure);
        }
    }
    
    public void Bar()
    {
    
    }
}
```

**Comment by meziantou** [comment]
The analyzer can't make such assomptions safely. You can disable the warning on specific cases.

Duplicates of #587 #613
